cache/lru/table.go

```diff
@@ -0,0 +1,67 @@
+package lru
+
+import (
+	"reflect"
+	"sync"
+
+	"github.com/golang/groupcache/lru"
+	"github.com/renproject/kv/db"
+)
+
+type lruTable struct {
+	mu    *sync.Mutex
+	lru   *lru.Cache
+	table db.Table
+}
+
+func NewLruTable(table db.Table, maxEntries int) db.Table {
+	return &lruTable{
+		mu:    new(sync.Mutex),
+		lru:   lru.New(maxEntries),
+		table: table,
+	}
+}
+
+func (table *lruTable) Insert(key string, value interface{}) error {
+	table.mu.Lock()
+	table.lru.Add(key, value)
+	table.mu.Unlock()
```

Comment:
Is there a reason the `Unlock` is not done in a `defer`? This is safer, because you do not necessarily know that `table.lru.Add` will not panic.